framework/encode/vulkan_state_writer.cpp

```diff
@@ -2344,47 +2344,49 @@ void VulkanStateWriter::WriteDescriptorUpdateCommand(format::HandleId
     const VkCopyDescriptorSet* copy = nullptr;
 
     // scratch-space for a potential pNext-struct
-    constexpr size_t max_num_bytes_p_next_data = std::max(sizeof(VkWriteDescriptorSetAccelerationStructureKHR),
-                                                          sizeof(VkWriteDescriptorSetInlineUniformBlockEXT));
+    constexpr size_t max_num_bytes_p_next_data =
+        std::max(sizeof(VkWriteDescriptorSetAccelerationStructureKHR), sizeof(VkWriteDescriptorSetInlineUniformBlock));
     std::array<uint8_t, max_num_bytes_p_next_data> p_next_data{};
 
+    write->pBufferInfo      = nullptr;
+    write->pImageInfo       = nullptr;
+    write->pTexelBufferView = nullptr;
+
     switch (write->descriptorType)
     {
         case VK_DESCRIPTOR_TYPE_SAMPLER:
         case VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER:
         case VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE:
         case VK_DESCRIPTOR_TYPE_STORAGE_IMAGE:
         case VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT:
-            write->pBufferInfo      = nullptr;
-            write->pImageInfo       = &binding->images[write->dstArrayElement];
-            write->pTexelBufferView = nullptr;
+            write->pImageInfo = &binding->images[write->dstArrayElement];
             break;
         case VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER:
         case VK_DESCRIPTOR_TYPE_STORAGE_BUFFER:
         case VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC:
         case VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC:
-            write->pBufferInfo      = &binding->buffers[write->dstArrayElement];
-            write->pImageInfo       = nullptr;
-            write->pTexelBufferView = nullptr;
+            write->pBufferInfo = &binding->buffers[write->dstArrayElement];
             break;
         case VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER:
         case VK_DESCRIPTOR_TYPE_STORAGE_TEXEL_BUFFER:
-            write->pBufferInfo      = nullptr;
-            write->pImageInfo       = nullptr;
             write->pTexelBufferView = &binding->texel_buffer_views[write->dstArrayElement];
             break;
-        case VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT:
-            // TODO
+        case VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK:
+            if (binding->inline_uniform_block != nullptr)
+            {
+                auto& p_next    = *reinterpret_cast<VkWriteDescriptorSetInlineUniformBlock*>(p_next_data.data());
+                p_next.sType    = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET_INLINE_UNIFORM_BLOCK;
+                p_next.pNext    = nullptr;
+                p_next.pData    = binding->inline_uniform_block.get();
+                p_next.dataSize = binding->count;
+                write->pNext    = &p_next;
+            }
```

Comment:
I wonder if it's possible the binding has both `inline_uniform_block` and `acceleration_structures`. This code is for that the binding has only `inline_uniform_block` or only `acceleration_structures`.

Reply:
It looks that it allows only one type. It's not a problem.